Tests/SonarQube.Common.UnitTests/SarifTests.cs

```diff
@@ -0,0 +1,293 @@
+//-----------------------------------------------------------------------
+// <copyright file="SarifTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Common.UnitTests
+{
+    [TestClass]
+    public class SarifTests
+    {
+        public TestContext TestContext { get; set; }
+
+        #region Tests
+
+        [TestMethod]
+        public void Utilities_SarifCompilerVersionCheck_IsUnsupported()
+        {
+            // Arrange
+            string testSarif = @"{
+  ""version"": ""0.1"",
+  ""toolInfo"": {
+                ""toolName"": ""Microsoft (R) Visual C# Compiler"",
+    ""productVersion"": ""1.0.0"",
+    ""fileVersion"": ""1.0.0""
+  }
+        }";
+
+            // Act
+            bool valid = SarifUtilities.IsSarifFromUnsupportedCompiler(testSarif);
+
+            // Assert
+            Assert.IsTrue(valid, "Expecting the compiler version check to return true for unsupported version");
```

Comment:
"Unsupported" is confusing word to use in this case. Changing the method name as suggested above would remove the confusion, or use a word like "handled" instead.